crates/turbopack-ecmascript/src/references/esm/url.rs

```diff
@@ -124,21 +124,42 @@ impl ChunkableModuleReference for UrlAssetReference {
 
 #[turbo_tasks::value_impl]
 impl CodeGenerateable for UrlAssetReference {
+    /// rewrites calls to the `new URL()` ctor depends on the current
+    /// conditions. This is based on the configuration (UrlRewriteBehavior),
+    /// and the current context (rendering), lastly the asset's condition
+    /// (if it's referenced / external). The following table shows the
+    /// behavior:
+    /*
+     * original call: `new URL(url, base);`
+    ┌───────────────────────────────┬─────────────────────────────────────────────────────────────────────────┬────────────────────────────────────────────────┬───────────────────────┐
+    │  UrlRewriteBehavior\RefAsset  │                         ReferencedAsset::Some()                         │           ReferencedAsset::External            │ ReferencedAsset::None │
+    ├───────────────────────────────┼─────────────────────────────────────────────────────────────────────────┼────────────────────────────────────────────────┼───────────────────────┤
+    │ Relative                      │ new URL(__turbopack_relative_url__(__turbopack_require__(urlId)), base) │ new URL(__turbopack_relative_url__(url), base) │ new URL(url, base)    │
+    │ Full(RenderingClient::Client) │ new URL(__turbopack_require__(urlId), "location.origin")                │ new URL(url, "location.origin")                │ new URL(url, base)    │
+    │ Full(RenderingClient::..)     │ new URL(__turbopack_resolve_module_id_path__(urlId))                    │ new URL(url, base)                             │ new URL(url, base)    │
```

Comment:
Why do we need another module wrapper function for this? Why can't we derive this from `CWD` or similar?

Reply:
We have cwd, but that's not composable with relative path url to its output asset.

The simpliest example is like `new URL('../../public/img.png', import.meta.url)`: we have correct import.meta.url points to yhe cwd of the module, but __turbopack_require_'d module points to the output distdir's asset with asset prefix so can't relate those two directlt.

This is the actual problem I struggled to solve without these wrapper, if there's a way to get a full phsyical path without asset prefix it'd be great. I feel I may miss something.